test/models/test_nn.py

```diff
@@ -64,23 +77,45 @@ def test_DenseSymm(symmetries, use_bias):
 @pytest.mark.parametrize("symmetries", ["trans", "autom"])
 @pytest.mark.parametrize("use_bias", [True, False])
 @pytest.mark.parametrize("lattice", [nk.graph.Chain, nk.graph.Square])
-def test_DenseEquivariant(symmetries, use_bias, lattice):
+@pytest.mark.parametrize("mode", ["fft", "matrix", "irrep"])
+def test_DenseEquivariant(symmetries, use_bias, lattice, mode):
     rng = nk.jax.PRNGSeq(0)
 
     g, hi, perms = _setup_symm(symmetries, N=3, lattice=lattice)
 
     pt = perms.product_table
     n_symm = np.asarray(perms).shape[0]
 
-    ma = nk.nn.DenseEquivariant(
-        symmetry_info=pt.ravel(),
-        in_features=1,
-        out_features=1,
-        use_bias=use_bias,
-        bias_init=nk.nn.initializers.uniform(),
-    )
+    if mode == "irrep":
+        ma = nk.nn.DenseEquivariant(
+            symmetries=perms,
+            mode=mode,
+            in_features=1,
+            out_features=1,
+            use_bias=use_bias,
+            bias_init=nk.nn.initializers.uniform(),
+        )
+    elif mode == "fft":
+        ma = nk.nn.DenseEquivariant(
+            symmetries=pt,
+            shape=tuple(g.extent),
+            mode=mode,
+            in_features=1,
+            out_features=1,
+            use_bias=use_bias,
+            bias_init=nk.nn.initializers.uniform(),
+        )
+    else:
+        ma = nk.nn.DenseEquivariant(
+            symmetries=pt,
+            mode=mode,
+            in_features=1,
+            out_features=1,
+            use_bias=use_bias,
+            bias_init=nk.nn.initializers.uniform(),
+        )
 
-    pars = ma.init(rng.next(), np.random.normal(0, 1, [1, n_symm]))
+    pars = ma.init(rng.next(), np.random.normal(0, 1, [1, 1, n_symm]))
```

Comment:
does this mean you also changed the shape of the input to the equivariant layer?

Reply:
Yes, both the output shape of DenseSymm and the input shape to DenseEquivariant have been changed to [n_samples, features, n_symm]